src/Simplify.elm

```diff
@@ -4108,6 +4135,54 @@ isNegatableOperator op =
             Nothing
 
 
+basicsToFloatChecks : CheckInfo -> Maybe (Error {})
+basicsToFloatChecks checkInfo =
+    case Evaluate.getNumber checkInfo checkInfo.firstArg of
+        Just _ ->
+            Just
+                (Rule.errorWithFix
+                    { message = "Unnecessary " ++ qualifiedToString (qualify checkInfo.fn defaultQualifyResources) ++ " on a literal number"
+                    , details =
+                        [ "A literal integers is considered as both an Int and a Float, there is therefore no need to explicitly convert it to a Float."
+                        , "You can replace this function call by the literal number."
+                        ]
+                    }
+                    checkInfo.fnRange
+                    (keepOnlyFix { parentRange = checkInfo.parentRange, keep = Node.range checkInfo.firstArg })
+                )
+
+        Nothing ->
+            Nothing
+
+
+intToIntChecks : CheckInfo -> Maybe (Error {})
+intToIntChecks =
+    firstThatConstructsJust
+        [ intToIntCheck
+        , onCallToInverseReturnsItsArgumentCheck Fn.Basics.toFloat
+        ]
+
+
+intToIntCheck : CheckInfo -> Maybe (Error {})
+intToIntCheck checkInfo =
+    case Evaluate.getInt checkInfo checkInfo.firstArg of
+        Just _ ->
+            Just
+                (Rule.errorWithFix
+                    { message = "Unnecessary integer conversion on a literal integer"
+                    , details =
+                        [ "Literal integers are already considered to be integers and it is therefore not necessary to convert them further."
```

Comment:
Maybe singular and say Int instead of integer? Something like

"A literal integer is already considered to be an Int which means converting it further is unnecessary."